src/cpp/handler/handlerengine.cpp

```diff
@@ -126,16 +126,13 @@ class InspectWorker : public Deferred
 	LastIds lastIds;
 	map<Deferred*, Connection> finishedConnection;
 
-	InspectWorker(ZrpcRequest *_req, ZrpcManager *_stateClient, bool _shareAll, QObject *parent = 0) :
-		Deferred(parent),
+	InspectWorker(ZrpcRequest *_req, ZrpcManager *_stateClient, bool _shareAll) :
 		req(_req),
 		stateClient(_stateClient),
 		shareAll(_shareAll),
 		truncated(false),
 		autoShare(false)
 	{
-		req->setParent(this);
```

Comment:
if this is removed then need to ensure `req` is cleaned up some other way

Reply:
thanks for catching this. no need to remove this line here.